tools/toolchains/__init__.py

```diff
@@ -911,6 +930,7 @@ def build_library(self, objects, dir, name):
 
         return needed_update
 
+    @check_toolchain_path
```

Comment:
@mazimkhan here is the decorator in use. This will cause an exception to be thrown before calling link_program if the path to the tool chain command doesn't exist.